tools/update_whatsnew.py

```diff
@@ -20,15 +21,15 @@
 
 # 1. Collect the whatsnew snippet files ---------------------------------------
 
-files = set(os.listdir(pr_dir))
+files = set()
+for f in glob(pjoin(pr_dir, '*.rst')):
+    files.add(f)
```

Comment:
I think I tried and this add the list to the set (got `List is unhashable` error).

I can try to unpack `set(*glob(pjoin(pr_dir, '*.rst')))`